lib/inputs/dovetail-frequency.js

```diff
@@ -0,0 +1,198 @@
+'use strict';
+
+const dynamo = require('../dynamo');
+const timestamp = require('../timestamp');
+const assayer = require('../assayer');
+const logger = require('../logger');
+
+// Assume the longest we keep data is 30 days
+// Use this to calculate ttl as updated_at + 30 days in seconds
+const FREQUENCY_HORIZON_SECONDS = 30 * 24 * 60 * 60; 
+
+/**
+ * Send dovetail impressions to ddb for frequency capped flights
+ */
+module.exports = class DovetailFrequency {
+  constructor(records) {
+    this._records = (records || []).filter(r => this.check(r));
+  }
+
+  check(record) {
+    return record.type === 'postbytes' &&
+      (record.impressions || []).length > 0 &&
+      this.checkImpressions(record.impressions);
+  }
+
+  checkImpressions(impressions) {
+    return impressions.some(this.checkFrequency);
+  }
+
+  checkFrequency(impression) {
+    if (impression === undefined ||
+        impression.frequency === undefined ||
+        impression.frequency === null) {
```

Comment:
Are these even possible?  I thought your DTR PR was always logging empty strings for every impression now.

Reply:
well, I am paranoid ;)
Also, as per your comments on that PR, I intend to change the DTR PR to only send frequency when there is a value, so yeah, I think this will happen